### DISCUSSION TITLE
AssistantModal v2

### DISCUSSION THREAD

**Discussion by Yonom** [assistant-ui/assistant-ui#235] [Ideas]
How can we improve AssistantModal?

**Comment by Yonom** [comment]
## Embeddability

Allow the creation of a JS snippet that you can give to your users so they can embed a modal chat interface on their website.

How might this work?

- iframe including the AssistantModal itself, with allowtransparency
  - TODO investigate downsides of this; e.g. can the user click "under" the transparent parts of the UI or are those clicks captured
- Embed FloatingAssistantButton, load iframe inside it
- Compile entire app and embed it without iframes

Let's take a look at other modal chat implementations (I'm sure there are a bunch of them OSS) to inform our decisionmaking.

**Reply by m13v** [reply]
I can push my code for modal iframe, you can test it on your end

**Comment by Yonom** [comment]
## Mobile Mode

AssistantModal should switch to a full-screen interface on mobile

**Reply by m13v** [reply]
makes sense, but i guess it still needs to appear as a chat window to avoid making it feel like a separate page

**Reply by Yonom** [reply]
Intercom

<img src="https://github.com/Yonom/assistant-ui/assets/1394504/12db7227-def3-4c66-b8d1-6067654b2d92" width="300px" />
<img src="https://github.com/Yonom/assistant-ui/assets/1394504/ef96b683-f8e5-49f1-9b87-86f864296f3d" width="300px" />

**Reply by m13v** [reply]
sounds good

**Comment by m13v** [comment]
i would start off by creating an example for an ecommerce website, where a user can link up their products and show them off in a carousel whenever a user asks for it, along with implementation of similar function calls

**Reply by Yonom** [reply]
Wonderful idea 👍
